src/reporting/disputeBondToken.se

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+#### must be first data members to match delegator.se
+data controller
+data controllerLookupName
+####
+
+inset('../macros/externs.sem')
+inset('../macros/safeMath.sem')
+inset('../macros/assertNoValue.sem')
+inset('../macros/require.sem')
+
+data initialized
+data market
+data bondHolder
+data payoutDistributionHash #Disputed outcome
+data bondRemainingToBePaidOut
+data tokens[2**160]
+data spenders[2**160][2**160]
+data supply
+
+def any():
+    assertNoValue()
+
+def initialize(market: address, bondHolder: address, bondAmount: int256, payoutDistributionHash: int256):
+    require(not self.initialized)
+    self.initialized = 1
+    self.market = market
+    self.bondHolder = bondHolder
+    self.payoutDistributionHash = payoutDistributionHash
+    self.bondRemainingToBePaidOut = bondAmount * 2
+    self.market.getReputationToken().transferFrom(msg.sender, self, bondAmount)
+
+def withdraw(amountToWithdraw: int256, payoutDistributionHash: int256):
+    require(msg.sender == self.bondHolder)
+    require(self.market.isFinalized())
+    require(payoutDistributionHash == self.payoutDistributionHash)
+    if (self.bondRemainingToBePaidOut > 0):
```

Comment:
This condition _shouldn't_ be necessary.  If the caller gets this far, any REP in this contract is theirs.  All we need to do is make sure that we track how much they have withdrawn (which you already do a few lines down) so that other parts of the code don't send too much REP to this contract.

`amountToTransfer` can just be set to `reputationToken.balanceOf(self)`.